magellan-legacy/src/main/java/com/wealthfront/magellan/Screen.java

```diff
@@ -71,75 +77,22 @@ public final Activity getActivity() {
   /**
    * @return the Navigator associated with this Screen.
    */
-  public final Navigator getNavigator() {
+  public final LegacyNavigator getNavigator() {
     return navigator;
   }
 
   public final Dialog getDialog() {
-    return dialog;
+    return dialogComponent.getDialog();
   }
 
-  final void restore(Bundle savedInstanceState) {
-    if (viewState == null && savedInstanceState != null) {
-      viewState = savedInstanceState.getSparseParcelableArray(VIEW_STATE + hashCode());
-    }
-  }
-
-  final V recreateView(Activity activity, Navigator navigator) {
-    this.activity = activity;
-    this.navigator = navigator;
-    view = createView(activity);
-    // noinspection unchecked
-    view.setScreen(this);
-    if (viewState != null) {
-      view.restoreHierarchyState(viewState);
-    }
-    return view;
-  }
-
-  final void save(Bundle outState) {
-    saveViewState();
-    if (viewState != null) {
-      outState.putSparseParcelableArray(VIEW_STATE + hashCode(), viewState);
-    }
-    viewState = null;
-  }
-
-  final void destroyView() {
-    saveViewState();
-    activity = null;
-    view = null;
-  }
-
-  private void saveViewState() {
-    if (view != null) {
-      viewState = new SparseArray<>();
-      view.saveHierarchyState(viewState);
-    }
-  }
-
-  final void createDialog() {
-    if (dialogCreator != null && dialogIsShowing) {
-      dialog = dialogCreator.createDialog(activity);
-      dialog.show();
-    }
-  }
-
-  final void destroyDialog() {
-    if (dialog != null) {
-      dialogIsShowing = dialog.isShowing();
-      dialog.setOnDismissListener(null);
-      dialog.dismiss();
-      dialog = null;
-    }
-  }
-
-  void transitionStarted() {
+  @Override
+  public void transitionStarted() {
     isTransitioning = true;
     transitionFinishedListeners.clear();
   }
 
-  void transitionFinished() {
+  @Override
+  public void transitionFinished() {
```

Comment:
If we're making this public, we should probably make it final too. The existing API for this is to use `whenTransitionFinished { }`

Reply:
d